src/overlays/actors/ovl_En_Bom/z_en_bom.c

```diff
@@ -21,19 +29,331 @@ const ActorInit En_Bom_InitVars = {
+            func_8002F850(globalCtx, &this->actor);
+            this->actor.velocity.y *= -0.3f;
+            this->actor.bgCheckFlags &= ~2;
+        } else if (this->timer >= 4) {
+            func_8002F580(&this->actor, globalCtx);
+        }
+    }
+
+    Actor_MoveForward(&this->actor);
+}
+
+void EnBom_WaitForRelease(EnBom* this, GlobalContext* globalCtx) {
+    // if attachedA is NULL bomb has been released
+    if (func_8002F5A0(&this->actor, globalCtx)) {
+        EnBom_SetupAction(this, EnBom_Move);
+        EnBom_Move(this, globalCtx);
+    }
+}
+
+void EnBom_Explode(EnBom* this, GlobalContext* globalCtx) {
+    Player* player;
+
+    if (this->jntSphList.list->dim.modelSphere.radius == 0) {
+        this->actor.flags |= 0x20;
+        func_800AA000(this->actor.xzDistFromLink, 0xFF, 0x14, 0x96);
+    }
+
+    this->jntSphList.list->dim.worldSphere.radius += this->actor.shape.rot.z + 8;
+
+    if (this->actor.params == 1) {
+        CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->jntSphList.base);
+    }
+
+    if (globalCtx->envCtx.unk_8C[3] != 0) {
+        globalCtx->envCtx.unk_8C[3] -= 0x19;
+    }
+
+    if (globalCtx->envCtx.unk_8C[4] != 0) {
+        globalCtx->envCtx.unk_8C[4] -= 0x19;
+    }
+
+    if (globalCtx->envCtx.unk_8C[5] != 0) {
+        globalCtx->envCtx.unk_8C[5] -= 0x19;
+    }
+
+    if (globalCtx->envCtx.unk_8C[0] != 0) {
+        globalCtx->envCtx.unk_8C[0] -= 0x19;
+    }
+
+    if (globalCtx->envCtx.unk_8C[1] != 0) {
+        globalCtx->envCtx.unk_8C[1] -= 0x19;
+    }
+
+    if (globalCtx->envCtx.unk_8C[2] != 0) {
+        globalCtx->envCtx.unk_8C[2] -= 0x19;
+    }
+
+    if (this->timer == 0) {
+        player = PLAYER;
+
+        if ((player->stateFlags1 & 0x800) && (player->heldActor == &this->actor)) {
+            player->actor.attachedB = NULL;
+            player->heldActor = NULL;
+            player->interactRangeActor = NULL;
+            player->stateFlags1 &= ~0x800;
+        }
+
+        Actor_Kill(&this->actor);
+    }
+}
+
+void EnBom_Update(Actor* thisx, GlobalContext* globalCtx) {
+    Vec3f effVelocity = { 0.0f, 0.0f, 0.0f };
+    Vec3f bomb2Accel = { 0.0f, 0.1f, 0.0f };
+    Vec3f effAccel = { 0.0f, 0.0f, 0.0f };
+    Vec3f effPos;
+    Vec3f dustAccel = { 0.0f, 0.6f, 0.0f };
+    Color_RGBA8_n dustColor = { 255, 255, 255, 255 };
+    s32 pad[2];
+    EnBom* this = THIS;
+
+    this->actor.gravity = -1.2f;
+
+    if (this->timer != 0) {
+        this->timer--;
+    }
+
+    if (this->timer == 67) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_PL_TAKE_OUT_SHIELD);
+        Actor_SetScale(&this->actor, 0.01f);
+    }
+
+    // activate bump collider if link is 20 or more units away from the bomb in the xz plane
+    // or if the height difference between link and the bomb is 80 or more units
+    if ((this->actor.xzDistFromLink >= 20.0f) || (ABS(this->actor.yDistFromLink) >= 80.0f)) {
+        this->bumpOn = true;
+    }
```

Comment:
I don't think these comments add any information that can't be obtained by simply reading the code, which is very clear to me with the current names.